kotlin-jvm/src/main/kotlin/playground/Apollo.kt

```diff
@@ -0,0 +1,33 @@
+@file:Suppress("PackageDirectoryMismatch")
+
+package playground.apollo
+
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.coroutines.await
+import com.apollographql.apollo.exception.ApolloException
+import kotlinx.coroutines.runBlocking
+import launches.GetLaunchesQuery
+
+
+fun main() {
+  println()
+  println("# apollographql/Apollo - Typesafe GraphQL in Kotlin")
+
+  val apolloClient = ApolloClient.builder()
+      .serverUrl("https://apollo-fullstack-tutorial.herokuapp.com/graphql")
+      .build()
+
+  runBlocking {
```

Comment:
I expected runBlocking to finish after all the println("Mission ...") are done, 
but that's not what happen on my mac,
the main function continues to run for like 1 minute before I get a 
"Process finished with exit code 0"

@LouisCAD any idea?

Reply:
@jmfayard Did you try adding a println call after the `runBlocking` block to see if it's something launched inside it or something elsewhere?

Reply:
> the main function continues to run for like 1 minute before I get a
> "Process finished with exit code 0"

I think this is because there are active threadpools (either apollo and/or okhttp)

Reply:
```
fun main() {
  runBlocking {
    // ....
    println("End of runBlocking()")
  }
  println("After runBlocking()")
}
```

does

```
Miision Starlink 1 launched from site CCAFS SLC 40 
End of runBlocking()
After runBlocking()
<WAIT ONE MINUTE>
Process finished with exit code 0
```

Reply:
So how do we close those threapools? :P

Reply:
> So how do we close those threapools? :P

I don't know :). Looking into it.

Reply:
I just pushed https://github.com/LouisCAD/kotlin-libraries-playground/pull/46/commits/113bec6aef71943978d20d1a3f7f60fcb4569e8b that fixes the issue for me. Turns out there were both Apollo **and** OkHttp threadpools active